orchestrator/utils/params/params.go

```diff
@@ -0,0 +1,46 @@
+package params
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	"github.com/SayedAlesawy/Videra-Ingestion/orchestrator/utils/errors"
+)
+
+// logPrefix Used for hierarchical logging
+var logPrefix = "[Params-Manager]"
+
+// orchestratorParamsOnce Used to garauntee thread safety for singleton instances
+var orchestratorParamsOnce sync.Once
+
+// orchestratorParamsInstance A singleton instance of the orchestrator params object
+var orchestratorParamsInstance *OrchestratorParams
+
+// OrchestratorParamsInstance A function to return the orchestrator params singleton instance
+func OrchestratorParamsInstance() *OrchestratorParams {
+	orchestratorParamsOnce.Do(func() {
+		params := os.Args
```

Comment:
I'm not sure if Go has this feature, but in python I can mark some args as required to be passed so they would automatically raise an error if not supplied

Reply:
This will panic if any of the parameters is missing anyway. I checked, there is a package called `flag` that I can use to flag command line params to look like this `./command-line-flags -word=opt -numb=7 -fork -svar=flag` it looks cleaner and more strict. Check this https://gobyexample.com/command-line-flags

Reply:
Yeah that will be great, the package nearly matches the behaviour I used in Python `argparse`

Reply:
Done.